Warning when compiling nanoarrow.hpp

Discovered when packaging the postgres driver:

```
../src/nanoarrow/nanoarrow.hpp:281:35: warning: field 'offset_' will be initialized after field 'arrays_' [-Wreorder-ctor]
      : EmptyArrayStream(schema), offset_(0), arrays_(std::move(arrays)) {}
                                  ^~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~
                                  arrays_(std::move(arrays)) offset_(0)
In file included from connection.cc:18:
```

--- a/src/nanoarrow/nanoarrow.hpp
+++ b/src/nanoarrow/nanoarrow.hpp
@@ -278,10 +278,10 @@ class VectorArrayStream : public EmptyArrayStream {
 
  protected:
   VectorArrayStream(struct ArrowSchema* schema, std::vector<UniqueArray> arrays)
-      : EmptyArrayStream(schema), offset_(0), arrays_(std::move(arrays)) {}
+      : EmptyArrayStream(schema), arrays_(std::move(arrays)), offset_(0) {}
 
   int get_next(struct ArrowArray* array) {
-    if (offset_ < arrays_.size()) {
+    if (offset_ < static_cast<int64_t>(arrays_.size())) {
       arrays_[offset_++].move(array);
     } else {
       array->release = nullptr;

